File: fe/travel-notes-app/src/app/core/models/note.ts
export interface NoteLocation {
  id: string;
  title: string;
  description: string;
  latitude: number;
  longitude: number;
  parentId: string | null;
  isArchived: boolean;
  childCount: number;
  createdAt: string;
  updatedAt: string;
}

export interface NoteLocationInput {
  title: string;
  description: string;
  latitude: number;
  longitude: number;
  parentId?: string | null;
  isArchived?: boolean;
}

export interface MapMarker {
  id: string;
  title: string;
  latitude: number;
  longitude: number;
  isArchived: boolean;
}

export interface Ancestor {
  id: string;
  title: string;
}

export interface SearchResult {
  note: NoteLocation;
  ancestors: Ancestor[];
}

// Temporary compatibility aliases for the existing flat-note UI. Phase 3 replaces its callers.
export type Note = NoteLocation;
export type NoteInput = NoteLocationInput;

File: be/travel-note-api/Models/Note.cs
namespace travel_note_api.Models;

public class Note
{
    public Guid Id { get; set; }
    public string Title { get; set; } = string.Empty;
    public string Description { get; set; } = string.Empty;
    public decimal Latitude { get; set; }
    public decimal Longitude { get; set; }
    public Guid? ParentId { get; set; }
    public Note? Parent { get; set; }
    public ICollection<Note> Children { get; set; } = new List<Note>();
    public bool IsArchived { get; set; }
    public DateTime CreatedAt { get; set; }
    public DateTime UpdatedAt { get; set; }
}
